packages/query-core/src/query.ts

```diff
@@ -130,6 +137,7 @@ export type Action<TData, TError> =
   | PauseAction
   | SetStateAction<TData, TError>
   | SuccessAction<TData>
+  | MetaAction
```

Comment:
`SetStateAction` has a weird property called `setStateOptions` I don't see anywhere it used, maybe its better to merge this action with `SetState` and `setStateOptions`

Reply:
yeah `setStateOptions` seems to be unsed! We could remove that I think :)

Reply:
How about putting meta in setStateOptions action and dispatch `setState`? I think its better to remove it.